docs/articles/multi-attribute-semantic-search.md

```diff
@@ -0,0 +1,483 @@
+
+Let's embark on our own quest to build this ultimate monster-finding system, using the power of multi-attribute vector search!
+
+## Dataset
+
+To demonstrate our search techniques, we'll use a small synthetic dataset generated by a Large Language Model (LLM). Here's the prompt used to create our monster database:
+
+```
+Generate two JSON lists: 'monsters' and 'queries'.
+
+1. 'monsters' list: Create 20 unique monsters with the following properties:
+   - name: A distinctive name
+   - look: Brief description of appearance (2-3 sentences)
+   - habitat: Where the monster lives (2-3 sentences)
+   - behavior: How the monster acts (2-3 sentences)
+
+   Ensure some monsters share similar features while remaining distinct.
+
+2. 'queries' list: Create 5 queries to search for monsters:
+   - Each query should be in the format: {look: "...", habitat: "...", behavior: "..."}
+   - Use simple, brief descriptions (1-3 words per field)
+   - Make queries somewhat general to match multiple monsters
+
+Output format:
+{
+  "monsters": [
+    {"name": "...", "look": "...", "habitat": "...", "behavior": "..."},
+    ...
+  ],
+  "queries": [
+    {"look": "...", "habitat": "...", "behavior": "..."},
+    ...
+  ]
+}
+```
+
+Let's examine a sample of our generated dataset. Note that your results may differ due to the non-deterministic nature of LLM generation.
+
+First five monsters:
+
+
+|    | name             | look                                                      | habitat                                             | behavior                                                      |
+|---:|:-----------------|:----------------------------------------------------------|:----------------------------------------------------|:--------------------------------------------------------------|
+|  0 | Luminoth         | Moth-like creature with glowing wings and antenna         | Dense forests and jungles with bioluminescent flora | Emits soothing light patterns to communicate and attract prey |
+|  1 | Aqua Wraith      | Translucent humanoid figure made of flowing water         | Rivers, lakes, and coastal areas                    | Shapeshifts to blend with water bodies and controls currents  |
+|  2 | Stoneheart Golem | Massive humanoid composed of interlocking rock formations | Rocky mountains and ancient ruins                   | Hibernates for centuries, awakens to protect its territory    |
+|  3 | Whispering Shade | Shadowy, amorphous being with glowing eyes                | Dark forests and abandoned buildings                | Feeds on fear and whispers unsettling truths                  |
+|  4 | Zephyr Dancer    | Graceful avian creature with iridescent feathers          | High mountain peaks and wind-swept plains           | Creates mesmerizing aerial displays to attract mates          |
+
```

Comment:
Is there a logic behind presenting the monsters (results) first, and then the queries, as opposed to the other way around?